auv_navigation/auv_localization/scripts/pressure_odometry_node.py

```diff
@@ -42,19 +46,75 @@ def __init__(self):
         rospy.loginfo(
             f"{pressure_odometry_colored} : depth covariance: {self.depth_calibration_covariance}"
         )
+
+        self.imu_data = None
+        self.imu_subscriber = rospy.Subscriber(
+            "imu/data", Imu, self.imu_callback, tcp_nodelay=True
+        )
+
         self.depth_subscriber = rospy.Subscriber(
             "depth", Float32, self.depth_callback, tcp_nodelay=True
         )
 
+        self.base_to_pressure_translation = None
+        rate = rospy.Rate(1.0)
+
+        while not rospy.is_shutdown() and self.base_to_pressure_translation is None:
+            try:
+                self.base_to_pressure_translation, _ = self.transformer.get_transform(
+                    "taluy/base_link", "taluy/base_link/external_pressure_sensor_link"
+                )
+            except Exception as e:
+                rospy.logwarn(f"Waiting for transform: {e}")
+                rate.sleep()
+
+    def imu_callback(self, imu_msg):
+        self.imu_data = imu_msg
+
+    def get_base_to_pressure_height(self):
+        try:
+            if self.base_to_pressure_translation is None:
+                rospy.logerr_throttle(10, "Transform not available.")
+                return 0.0
+
+            if self.imu_data is None:
+                rospy.logwarn_throttle(
+                    10, "No IMU data received yet. Using default orientation."
+                )
+                return self.base_to_pressure_translation[0, 2]
+
+            else:
```

Comment:
this `else` is not required since the above if block does an early return

**prefer this**
```py
def some_function(self):
    if some_condition:
        do_something()
        return something
   
    do_something_else()
    return something_else
```

**over this**
```py
def some_function(self):
    if some_condition:
        do_something()
        return something
    else:
       do_something_else()
       return something_else
```